### ISSUE TITLE
Enhancment

### ISSUE THREAD

**Issue by iplayfast** [LAION-AI/Open-Assistant#2601]
I don't know if this is possible, but I would like the ability to upload a file or set of files, then use open-assist to manipulate them and then download them once the manipulation is done.

For example, I've got code that has a bug, but it is too large to simply copy/paste into chat, and I'd like assistance debugging it. 
Or I need to develop some specialized class open-assist can do the typing for me, and discuss problems or ideas in the code. 

Or I've written a book, and want open-assist to point out weak areas, or give new ideas to enhance the work.

You get the idea. I want open-assistant to assist me with documents of all types.

**Comment by olliestanley** [comment]
Just like ChatGPT or any model there is a limit on the amount of "context" the model can consider at once. This means it isn't possible to directly work with a large document like a book or a codebase.

The OA team has some people who are working on a plugins system via which we hope to implement things like retrieval which will hopefully make progress towards use cases like yours, so keep an eye out!

**Comment by iplayfast** [comment]
Yes, the book was the impossible wish, (but still I wish it). I would be happy if header and cpp files could be examined or modified using chat. I like the idea of me explaining what I want, and the computer doing it. Then I refine what it did and so on.

**Comment by andreaskoepf** [comment]
The primary problem here is not a technical .. for example a vector db + retrieval could be used to find relevant parts of a book or larger document. But this is a non-trivial extension, e.g. we need someone who is a retrieval expert and has 160h+ time to implement & test this potentially together with people from frontend team. If someone is interested please let me know .. it would definitely be a super cool feature. And as Ollie mentioned context size is also a related topic .. we currently work with 2k token context size, ideally this should be increased to a larger context size. But it we need corresponding attention-kernels + longcontext-training.
